auth/src/main/java/io/grpc/auth/GoogleAuthLibraryCallCredentials.java

```diff
@@ -353,4 +357,30 @@ public Credentials tryServiceAccountToJwt(Credentials creds) {
       return creds;
     }
   }
+
+  /**
+   * This method is to support the hack for AppEngineCredentials which need to run on a
+   * specific thread.
+   * @return Whether a specific executor is needed or if any executor can be used
+   */
+  @Override
+  public boolean isSpecificExecutorRequired() {
+    // Cache the value so we only need to try to load the class once
+    if (requiresSpecificExecutor == null) {
+      if (googleCredentialsClass == null) {
+        requiresSpecificExecutor = Boolean.FALSE;
+      } else {
+        try {
+          requiresSpecificExecutor =
+              Class.forName("com.google.auth.oauth2.AppEngineCredentials")
```

Comment:
We shouldn't do the `Class.forName()` repeatedly. We should load the class once and save it statically. The common case is the class won't be present and we don't want to cause an exception potentially per-RPC (e.g., if the application is setting the credentials per-RPC).

Reply:
I was thinking that the credentials object would be fairly persistent so that caching the value would eliminate the repeated invocations.  Even if that had been true, it would be cleaner to store it statically.